vm/src/stdlib/os.rs

```diff
@@ -159,6 +159,22 @@ impl TryFromObject for PyPathLike {
     }
 }
 
+fn path_from_fd(raw_fd: i64) -> Result<PathBuf, String> {
+    cfg_if::cfg_if! {
+        if #[cfg(any(target_os = "linux", target_os = "macos", windows))] {
+            let path = match rust_file(raw_fd).path() {
+                Ok(path) => path,
+                Err(_) => {
+                    return Err(format!("Cannot determine path of fd: {:?}", raw_fd));
+                }
+            };
+            Ok(path)
+        } else {
+            Err(vm.new_os_error("fd not supported on wasi yet".to_owned()));
+        }
+    }
```

Comment:
This won't compile on wasi I think - it tries to call something on vm and it returns a py exception instead of a string - but I think it's fine to just accept `vm` as an argument and return `PyResult<PathBuf>`

Reply:
I remove the `vm` in `path_from_fd` and let the caller decide what to do with the error (`String`).